src/map/utils/charutils.cpp

```diff
@@ -766,6 +759,61 @@ namespace charutils
         luautils::OnGameIn(PChar, zoning == 1);
     }
 
+    void LoadSpells(CCharEntity* PChar)
+    {
+        // disable all spells
+        PChar->m_SpellList.reset();
+        PChar->m_EnabledSpellList.reset();
+
+        // Compile a string of all enabled expansions
+        std::vector<const char*> expansions {"COP", "TOAU", "WOTG", "ACP", "AMK", "ASA", "ABYSSEA", "SOA"};
+        std::string enabledExpansions;
+
+        uint8 count = 0;
+        for (const char* expan : expansions)
+        {
+            if (luautils::IsExpansionEnabled(expan))
+            {
+                if (count > 0)
+                {
+                    enabledExpansions += ",";
+                }
+                enabledExpansions += "\"";
+                enabledExpansions += expan;
+                enabledExpansions += "\"";
+                count++;
+            }
+        }
+
+        // Select all player spells from enabled expansions
+        const int8* fmtQuery =
+            "SELECT char_spells.spellid "
+            "FROM char_spells "
+            "JOIN spell_list "
+            "ON spell_list.spellid = char_spells.spellid "
+            "WHERE charid = %u AND "
+            "(spell_list.required_expansion IN (%s) OR "
+            "spell_list.required_expansion IS NULL);";
+
+        int32 ret = Sql_Query(SqlHandle, fmtQuery, PChar->id, enabledExpansions.c_str());
+
+        if (ret != SQL_ERROR &&
+            Sql_NumRows(SqlHandle) != 0)
+        {
+            while(Sql_NextRow(SqlHandle) == SQL_SUCCESS)
+            {
+                uint16 spellId = Sql_GetUIntData(SqlHandle, 0);
+
+                if (spell::GetSpell(spellId) != nullptr)
+                {
+                    PChar->m_SpellList[spellId] = true;
+                    PChar->m_EnabledSpellList[spellId] = true;
```

Comment:
get rid of EnabledSpellList and just replace everywhere it was used with SpellList